### PR TITLE
#1345 Add list of most recent donations

### PR DESCRIPTION
**Pull request by johnmgrimm**
Web3 is not capable of listing most recent transactions for the selected address so alternative approach was taken which is using free Covalent API that lets to make complex queries to the Ethereum blockchain.

Two configuration options added donationAddress and covalentApiKey.

Here is how the list looks:

image

### CODE REVIEW

File: src/main/resources/config.properties [hunk 1]
```diff
@@ -13,3 +13,10 @@ pom.version=${project.version}
 # The language of the educational content
 # See https://github.com/elimu-ai/model/blob/master/src/main/java/ai/elimu/model/enums/Language.java
 content.language=ENG
+
+# Create free API key using https://www.covalenthq.com/platform/#/auth/register/
+# It will be used to get ETH transaction data from https://www.covalenthq.com/docs/api/#get-/v1/{chain_id}/address/{address}/transactions_v2/
+covalentApiKey=USE_YOUR_API_KEY
+
+# Used to fetch latest transactions
+donationAddress=0x883753Beab357A2c29f3766C6ad158e72A78ce51
```
Comment (nya-elimuai) [inline src/main/resources/config.properties:22]: Add newline character:

```suggestion
donationAddress=0x883753Beab357A2c29f3766C6ad158e72A78ce51

```

File: src/main/webapp/WEB-INF/jsp/contributions/donate.jsp [hunk 2]
```diff
@@ -247,4 +247,26 @@
             </script>
         </div>
     </div>
+
+    <!-- Get donations list using Covalent API -->
+    <script type="text/javascript">
```
Comment (nya-elimuai) [inline src/main/webapp/WEB-INF/jsp/contributions/donate.jsp:252]: The `type` attribute should be [omitted](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-type) in HTML5:

```suggestion
    <script>
```

File: src/main/webapp/WEB-INF/jsp/contributions/donate.jsp [hunk 3]
```diff
@@ -247,4 +247,26 @@
...
+        var url = 'https://api.covalenthq.com/v1/1/address/${applicationScope.configProperties['donationAddress']}/transactions_v2/?no-logs=true&limit=10&match=%7B%22to_address%22%3A+%22${applicationScope.configProperties['donationAddress']}%22%2C+%22value%22%3A+%7B+%22%24gt%22%3A+0+%7D++%7D&key=${applicationScope.configProperties['covalentApiKey']}';
```
Comment (nya-elimuai) [inline src/main/webapp/WEB-INF/jsp/contributions/donate.jsp:253]: Does Covalent also support the Rinkeby testnet?

File: src/main/webapp/static/css/styles.css [hunk 4]
```diff
@@ -207,3 +207,14 @@ form {
...
+.donations-list {
+    margin-bottom: 20px;
+}
+
+.donation-list .donation-address {
+    max-width: 200px;
+    text-overflow: ellipsis;
+    display: inline-block;
+    overflow: hidden;
+}
```
Comment (nya-elimuai) [inline src/main/webapp/static/css/styles.css:220]: Add newline character:

```suggestion
}

```

File: src/main/webapp/static/css/styles.css [hunk 5]
```diff
@@ -207,3 +207,14 @@ form {
 #formLoadingOverlay > .preloader-wrapper {
     top: 50%;
 }
+
+.donations-list {
```
Comment (nya-elimuai) [inline src/main/webapp/static/css/styles.css:211]: Typo:

```suggestion
.donation-list {
```

File: src/main/webapp/static/css/styles.css [hunk 6]
```diff
@@ -207,3 +207,14 @@ form {
...
+    margin-bottom: 20px;
```
Comment (nya-elimuai) [inline src/main/webapp/static/css/styles.css:212]: Use relative margin instead of fixed margin:

```suggestion
    margin-bottom: 1rem;
```

**Review by nya-elimuai — CHANGES_REQUESTED** [review]
Tested locally, and it works. However, it fetches data from the mainnet, and not from the testnet. Would it be possible to fetch the data from the Rinkeby testnet during development/testing?

**Comment by nya-elimuai** [comment]
@johnmgrimm Also, I pushed some server config changes onto your branch: https://github.com/johnmgrimm/webapp/commit/7a929dc51387d93ab7723ec471fc5e46178659ec

File: src/main/webapp/WEB-INF/jsp/contributions/donate.jsp (hunk 3 above)
Comment (johnmgrimm) [reply src/main/webapp/WEB-INF/jsp/contributions/donate.jsp]: Unfortunately it doesn't support Rinkeby but it does support Kovan (https://www.covalenthq.com/docs/networks#testnets).

I can create a test setup with Kovan if you like.

Comment (nya-elimuai) [reply src/main/webapp/WEB-INF/jsp/contributions/donate.jsp]: > Unfortunately it doesn't support Rinkeby but it does support Kovan (https://www.covalenthq.com/docs/networks#testnets).
> 
> I can create a test setup with Kovan if you like.

I see. Then let's stick with the current solution (Ethereum Mainnet in both `TEST` and `PROD`).

**Review by nya-elimuai — APPROVED** [review]
Ready to be merged 👍

**Pull request merged**